.clang-format

```diff
@@ -0,0 +1,13 @@
+---
+BasedOnStyle: Mozilla
+
+---
+Language: Cpp
+
+# Avoid excessive linebreaks on function definitions
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+
+IndentWidth: 4
+DerivePointerAlignment: false
+---
```

Comment:
For some reason the formatting rules were different in this repo so all the new code was failing. Adding this fixes it but changes the formatting in other files. I'll comment where relevant.